backend/capellacollab/settings/modelsources/t4c/routes.py

```diff
@@ -77,9 +78,12 @@ def edit_t4c_instance(
     instance: DatabaseT4CInstance = Depends(get_existing_instance),
     db: Session = Depends(get_db),
 ) -> DatabaseT4CInstance:
-    for key in body.dict():
+    for key in FieldsT4CInstance(**body.dict()).dict():
         if value := body.__getattribute__(key):
             instance.__setattr__(key, value)
+    if body.version_id:
+        version = get_version_by_id_or_raise(db, body.version_id)
+        instance.version = version
```

Comment:
Can't you just update the `version_id`. This would make the changes redundant.

Reply:
You don't have to assign the `version` variable, instead just call the method. Apart from that, the suggestion looks good.

Reply:
Edited and commited!